nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,608 @@
+        f"'icephys_experiment_type' should be 'voltage_clamp', 'current_clamp' or 'izero', but received value "
+        f"{icephys_experiment_type}"
+    )
+
+    # Check and auto-create electrodes, in case they don't existe yet on nwbfile
+    if len(nwbfile.ic_electrodes) == 0:
+        warnings.warn(
+            "When adding Icephys Recording, no Icephys Electrodes were found on nwbfile. Creating Electrodes now..."
+        )
+        add_icephys_electrode(
+            neo_reader=neo_reader,
+            nwbfile=nwbfile,
+            metadata=metadata,
+        )
+
+    if getattr(nwbfile, "intracellular_recordings", None):
+        ri = max(nwbfile.intracellular_recordings["responses"].index)
+    else:
+        ri = -1
+    if getattr(nwbfile, "icephys_simultaneous_recordings", None):
+        simultaneous_recordings_offset = len(nwbfile.icephys_simultaneous_recordings)
+    else:
+        simultaneous_recordings_offset = 0
+
+    if getattr(nwbfile, "icephys_sequential_recordings", None):
+        sessions_offset = len(nwbfile.icephys_sequential_recordings)
+    else:
+        sessions_offset = 0
+
+    relative_session_start_time = metadata["Icephys"]["Sessions"][sessions_offset]["relative_session_start_time"]
+    session_stimulus_type = metadata["Icephys"]["Sessions"][sessions_offset]["stimulus_type"]
+
+    # Sequential icephys recordings
+    simultaneous_recordings = list()
+    for si in range(n_segments):
+        # Parallel icephys recordings
+        recordings = list()
+        for ei, electrode in enumerate(
+            list(nwbfile.icephys_electrodes.values())[: len(neo_reader.header["signal_channels"]["units"])]
+        ):
+            # Starting time is the signal starting time within .abf file + time
+            # relative to first session (first .abf file)
+            ri += 1
+            starting_time = neo_reader.get_signal_t_start(block_index=0, seg_index=si)
+            starting_time = starting_time + relative_session_start_time
+
+            sampling_rate = neo_reader.get_signal_sampling_rate()
+            response_unit = neo_reader.header["signal_channels"]["units"][ei]
+            response_conversion = get_conversion_from_unit(unit=response_unit)
+            response_gain = neo_reader.header["signal_channels"]["gain"][ei]
+            response_name = f"{icephys_experiment_type}-response-{si + 1 + simultaneous_recordings_offset}-ch-{ei}"
+
+            response = response_classes[icephys_experiment_type](
+                name=response_name,
+                description=f"Response to: {session_stimulus_type}",
+                electrode=electrode,
+                data=neo_reader.get_analogsignal_chunk(block_index=0, seg_index=si, channel_indexes=ei),
+                starting_time=starting_time,
+                rate=sampling_rate,
+                conversion=response_conversion * response_gain,
+                gain=1.0,
```

Comment:
Default `gain` for the NWB level response/stimulus should be `np.nan` to clarify that it was not set. It's a common issue otherwise to see a gain of `1.0` but data stored as some `int` type.

Reply:
It's also fairly confusing from the documentation of https://pynwb.readthedocs.io/en/stable/pynwb.icephys.html#pynwb.icephys.PatchClampSeries as to why there are separate fields for `gain` and `conversion`. Do you know what that is about? Typically the `conversion` factor is all one needs